subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/NoHistoryArtifactState.java

```diff
@@ -80,4 +82,9 @@ public FileCollection getOutputFiles() {
     public OverlappingOutputs getOverlappingOutputs() {
         return null;
     }
+
+    @Override
+    public Set<File> getPreviousOutputs() {
+        return null;
+    }
```

Comment:
I think it's fine as it is. We probably shouldn't adopt `Optional` before upgrading to Java 8 in any case.